src/elevation/getElevation.ts

```diff
@@ -0,0 +1,58 @@
+import type {Position} from 'geojson';
+import {fetchElevationTileData} from './fetchElevationTileData.js';
+import {ElevationTileData, NO_DATA} from './shared.js';
+
+// This file is largely derived from https://github.com/perliedman/node-hgt/blob/master/src/hgt.js
+export async function getElevation(coord: Position) {
+    try {
+        const hgt = await fetchElevationTileData(coord);
+
+        const size = hgt.size - 1;
+
+        const row = (coord[1] - hgt.swLngLat[1]) * size;
+        const col = (coord[0] - hgt.swLngLat[0]) * size;
+
+        if (row < 0 || col < 0 || row > size || col > size) {
+            throw `Lat/lng is outside tile bounds (row=${row}, col=${col}; size=${size})`;
+        }
+
+        // This could be changed to nearestNeighbor or configured with options,
+        // for example return hgt.options.interpolation.call(this, row, col);
+        return bilinear(hgt, row, col);
+    } catch (e: unknown) {
+        // treat `NO_DATA` as sea level
+        if (e === NO_DATA) return 0;
```

Comment:
💯

Reply:
do we want to also `return 0` for `e === TILE_MISSING`?

Reply:
`fetchElevationTileData` traps the `TILE_MISSING` exception, and returns `NO_DATA` instead.

1. `fetchTileData` throws `TILE_MISSING`
2. `fetchElevationTileData` catches `TILE_MISSING` and throws `NO_DATA` instead
3. `getElevation` catches `NO_DATA` and **_returns_** `0` as the elevation.